### PR TITLE
Shopingcart

### PR DESCRIPTION
**Pull request by shaileshdeuex**
created e-commerce store with Fake store API

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,25 +1,141 @@
...
+    setAppState({ loading: true, data: [] });
+    const apiUrl = "https://fakestoreapi.com/products";
+    fetch(apiUrl)
+      .then((res) => res.json())
+      .then(async (shopingData) =>
+        setAppState({ loading: false, data: shopingData })
+      )
+      .catch((err) => {
+        console.log(err);
+        setAppState({ loading: false, data: [] });
```
Comment (shahsank3t) [inline src/App.tsx]: We should show some error to user so they know that there was some error in fetching and not that there are no data to be displayed

File: src/App.tsx [hunk 2]
```diff
@@ -1,25 +1,141 @@
...
+      });
+  }, [setAppState]);
+
+  const cartIncluds = (id: number) => {
+    let includs = false;
+    for (const cartData of cart) {
```
Comment (shahsank3t) [inline src/App.tsx]: instead of `for` loop, you can use `.find` method of array.

File: src/App.tsx [hunk 3]
```diff
@@ -1,25 +1,141 @@
...
+export type ShoppingCart = {
+  id: number;
+  itemQty: number;
+  itemName: string;
+  image: string;
+  price: number;
+};
 
 function App() {
+  const [sideDrawer, setsideDrawer] = useState(false);
```
Comment (shahsank3t) [inline src/App.tsx]: `setsideDrawer` should be proper camelcase `setSideDrawer`

File: src/App.tsx [hunk 4]
```diff
@@ -1,25 +1,141 @@
...
+  const [cart, setcart] = useState([] as ShoppingCart[]);
```
Comment (shahsank3t) [inline src/App.tsx]: same here, `setCart`

**Review by shahsank3t — CHANGES_REQUESTED** [review]
There should also be a close icon on the Drawer that comes up on clicking the cart icon. Currently, there's no way I can clearly say how to close the drawer.

File: src/App.tsx (hunk 1 above)
Comment (shaileshdeuex) [reply src/App.tsx]: Implemented

File: src/App.tsx (hunk 2 above)
Comment (shaileshdeuex) [reply src/App.tsx]: Implemented

File: src/App.tsx (hunk 3 above)
Comment (shaileshdeuex) [reply src/App.tsx]: Implemented

File: src/App.tsx (hunk 4 above)
Comment (shaileshdeuex) [reply src/App.tsx]: Implemented

**Review by shahsank3t — APPROVED** [review]
+1

**Pull request merged**